XLS-47d-PriceOracles/README.md

```diff
@@ -0,0 +1,468 @@
+- `base_asset` or `quote_asset` are missing.
+- `trim` or `time_threshold` contain invalid uint value.
+- If the resulting data set is empty.
+
+###### Example of get_aggregate_price API JSON request
+
+    {
+    "method": "get_aggregate_price",
+    "params": [
+        {
+            "ledger_index": "current",
+            "base_asset": "XRP",
+            "quote_asset": "USD",
+            "flags": 7,
+            "trim": 20,
+            "oracles": [
+              {
+                "account": "rp047ow9WcPmnNpVHMQV5A4BF6vaL9Abm6,
+                "oracle_sequence": 34
+              },
+              {
+                "account": "rp147ow9WcPmnNpVHMQV5A4BF6vaL9Abm7,
+                "oracle_sequence": 56
+              },
+              {
+                "account": "rp247ow9WcPmnNpVHMQV5A4BF6vaL9Abm8,
+                "oracle_sequence": 2
+              },
+              {
+                "account": "rp347ow9WcPmnNpVHMQV5A4BF6vaL9Abm9,
+                "oracle_sequence": 7
+              },
+              {
+                "account": "rp447ow9WcPmnNpVHMQV5A4BF6vaL9Abm0,
+                "oracle_sequence": 109
+              }
+            ]
+        }
+    ]
+    }
+
+##### Output fields
+
+On success, the response data contains the following fields:
+
+- `entire_set` is an object of the following fields:
+  - `size` is the size of the data set used to calculate the statistics.
+  - `average` is the simple average.
+  - `standard_deviation` is the standard deviation.
+- `trimmed_set` is an object, which is included in the response if `trim` fields is set. The object has the following fields:
+  - `size` is the size of the data set used to calculate the statistics.
+  - `average` is the simple average.
+  - `standard_deviation` is the standard deviation.
+- `median` is the median.
+- `time` is the most recent time stamp out of all `LastUpdateTime` values.
+
+###### Example of get_aggregate_price API JSON response
+
+    {
+      "entire_set" : {
+         "average" : "74.75",
+         "size" : 10,
+         "standard_deviation" : "0.1290994448735806"
+      },
+      "ledger_current_index" : 25,
+      "median" : "74.75",
+      "status" : "success",
+      "trimmed_set" : {
+        "average" : "74.75",
+        "size" : 6,
+        "standard_deviation" : "0.1290994448735806"
+      },
+      "validated" : false
+      "time" : 78937648
+    }
+
+## Appendices
+
+### Appendix 1. STI_CURRENCY
+
+A new type, `STI_CURRENCY`, is introduced to support `BaseAsset` and `QuoteAsset` fields' type `CURRENCY`. This type can represent a standard currency code or an arbitrary asset as a 160-bit (40 character) hexadecimal string. This type is conformant to the XRPL [Currency Codes](https://xrpl.org/currency-formats.html#currency-codes). Below is a JSON example with the `BaseAsset` representing a CUSIP code `912810RR9` as a 160-bit hexadecimal string and a `QuoteAsset` representing a standard `USD` currency code:
```

Comment:
https://xrpl.org/currency-formats.html#currency-codes explicitly disallows XRP, it would be good to mention that in this context XRP is allowed.